tests/virt_autotest/virt_utils.pm

```diff
@@ -379,4 +380,28 @@ sub is_installed_equal_upgrade_major_release {
     return $host_installed_version eq $host_upgrade_version;
 }
 
+#This subroutine is used for generating junit log, especially for those test modules outside virt_autotest
+#or use other bases instead of virt_autotest_base
+sub make_junit_log {
+    my ($self, $status, $runtime, $product_name, $package_name) = @_;
+    my $tc_result = virt_autotest_base::analyzeResult($self, $status, $runtime);
+    $self->{"product_tested_on"} = script_output("cat /etc/issue | grep -io \"SUSE.*\$(arch))\"");
+    $self->{"product_name"}      = $product_name;
+    $self->{"package_name"}      = $package_name;
+    @{$self->{success_guest_list}} = keys %xen::guests;
+    my $xml_result = virt_autotest_base::generateXML($self, $tc_result);
+    # Upload and parse junit file.
+    virt_autotest_base::push_junit_log($self, $xml_result);
+}
+
+#This subroutine is used for generating time duration in format like xhymzs
+sub duration_calc_in_hms {
+    my ($self, $time_duration) = @_;
+    my $time_duration_hours = int($time_duration / 3600);
+    my $time_duration_mins  = int(($time_duration - ($time_duration_hours * 3600)) / 60);
```

Comment:
something flaky happening here , so if `$time_duration_hours` is equal to `$time_duration / 3600` this basically means that 
`$time_duration_mins = (( $time_duration - ( $time_duration /3600 * 3600 )) / 60 );`

which means that  :
`$time_duration_mins = (( $time_duration - $time_duration )) / 60 );`

which means that : 
`$time_duration_mins = 0;`

Reply:
More exactly, 
$time_duration_mins = int(( $time_duration - ( int($time_duration /3600) * 3600 )) / 60 );
So it can be deduced that $time_duration_mins=0
int($time_duration /3600)  is smaller than ($time_duration /3600)

Reply:
you right I have miss that . But you can replace whole function with one system call :

`strftime("\%H:\%M:\%S", gmtime($time_duration));'`

Reply:
This is what I tried to avoid in the first place, because not all built-in subroutines can be called directly. It is inevitable to include specific modules before using them. And sometimes these modules are not standard ones. So which module provides "strftime" ? And is this module installed and included in @INC ?

Reply:
It seems it is  use POSIX 'strftime'; Can I use this directly ?

Reply:
yes I think so , but I haven't tried .

Reply:
I will give it a try.